src/libs/actions/Session/index.js

```diff
@@ -164,59 +144,40 @@ function beginSignIn(login) {
-    const autoGeneratedLogin = Str.guid('expensify.cash-');
-    const autoGeneratedPassword = Str.guid();
+function signInWithShortLivedAuthToken(authToken) {
+    const optimisticData = [
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: ONYXKEYS.ACCOUNT,
+            value: {
+                ...CONST.DEFAULT_ACCOUNT_DATA,
+                isLoading: true,
+            },
+        },
+    ];
 
-    return DeprecatedAPI.CreateLogin({
-        authToken,
-        partnerName: CONFIG.EXPENSIFY.PARTNER_NAME,
-        partnerPassword: CONFIG.EXPENSIFY.PARTNER_PASSWORD,
-        partnerUserID: autoGeneratedLogin,
-        partnerUserSecret: autoGeneratedPassword,
-        shouldRetry: false,
-        forceNetworkRequest: true,
-        email,
-        includeEncryptedAuthToken: true,
-    })
-        .then((createLoginResponse) => {
-            if (createLoginResponse.jsonCode !== 200) {
-                Onyx.merge(ONYXKEYS.ACCOUNT, {errors: {[DateUtils.getMicroseconds()]: Localize.translate('createLoginResponse.message')}});
-                return createLoginResponse;
-            }
-
-            setSuccessfulSignInData(createLoginResponse);
-
-            // If we have an old generated login for some reason
-            // we should delete it before storing the new details
-            if (credentials && credentials.autoGeneratedLogin) {
-                DeprecatedAPI.DeleteLogin({
-                    partnerUserID: credentials.autoGeneratedLogin,
-                    partnerName: CONFIG.EXPENSIFY.PARTNER_NAME,
-                    partnerPassword: CONFIG.EXPENSIFY.PARTNER_PASSWORD,
-                    shouldRetry: false,
-                })
-                    .then((response) => {
-                        if (response.jsonCode === CONST.JSON_CODE.SUCCESS) {
-                            return;
-                        }
-
-                        Log.hmmm('[Session] Unable to delete login', false, {message: response.message, jsonCode: response.jsonCode});
-                    });
-            }
-
-            Onyx.merge(ONYXKEYS.CREDENTIALS, {
-                autoGeneratedLogin,
-                autoGeneratedPassword,
-            });
-            return createLoginResponse;
-        })
-        .finally(() => {
-            Onyx.merge(ONYXKEYS.ACCOUNT, {isLoading: false});
-        });
+    const successData = [
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: ONYXKEYS.ACCOUNT,
+            value: {
+                isLoading: false,
+            },
+        },
+    ];
+
+    const failureData = [
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: ONYXKEYS.ACCOUNT,
+            value: {
+                isLoading: false,
+            },
+        },
+    ];
+
+    API.write('SignInWithShortLivedAuthToken', {authToken, oldPartnerUserID: credentials ? credentials.autoGeneratedLogin : ''}, {optimisticData, successData, failureData});
```

Comment:
```suggestion
    API.write('SignInWithShortLivedAuthToken', {authToken, oldPartnerUserID: credentials.autoGeneratedLogin || ''}, {optimisticData, successData, failureData});
```

Reply:
@iwiznia, had to change this back to the ternary as the `crendetials.autoGeneratedLogin || ''` resulted in the error @aimane-chnaif experienced [here](https://github.com/Expensify/App/pull/13456#issuecomment-1356327312).

Reply:
hmmmm odd, I think it's because even though we initialize it to `{}` we then set it to `null`. Given I see other places where we use credentials without checking it's an object first. Do you think we should change it so credentials is always an object?
I think that would be [here](https://github.com/Expensify/App/blob/2878072dc0096fece156d69189f77bbca46b80d5/src/libs/actions/Session/index.js#L62) and [here](https://github.com/Expensify/App/blob/2878072dc0096fece156d69189f77bbca46b80d5/src/libs/actions/Session/index.js#L364) or maybe just [here](https://github.com/Expensify/App/blob/2878072dc0096fece156d69189f77bbca46b80d5/src/libs/actions/Session/index.js#L27)

Reply:
Found a couple other places but I don't mind updating it, since it probably should always be an object.

Reply:
@iwiznia updated!